plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+            self.module, self.idrac, MANAGERS_URI)
+        if error_msg:
+            self.module.exit_json(msg=error_msg, failed=True)
+        resp = get_dynamic_uri(self.idrac, uri)
+        url = resp.get('Links', {}).get(OEM, {}).get(MANUFACTURER, {}).get(LICENSE_MANAGEMENT_SERVICE, {}).get('@odata.id', {})
+        action_resp = get_dynamic_uri(self.idrac, url)
+        if module.params.get('share_parameters').get('share_type') == "local":
+            license_service = IMPORT_LOCAL
+        else:
+            license_service = IMPORT_NETWORK_SHARE
+        import_url = action_resp.get(ACTIONS, {}).get(license_service, {}).get('target', {})
+        return import_url
+
+
+def check_license_id(self, module, license_id, operation):
+    license_uri = get_license_url(self)
+    license_url = license_uri + f"/{license_id}"
+    try:
+        response = self.idrac.invoke_request(license_url, 'GET')
+        return response
+    except Exception:
+        module.exit_json(msg=FAILURE_MSG.format(operation=operation, license_id=license_id), failed=True)
+
+
+def get_license_url(self):
+    v1_resp = get_dynamic_uri(self.idrac, REDFISH)
+    license_service_url = v1_resp.get('LicenseService', {}).get('@odata.id', {})
+    license_service_resp = get_dynamic_uri(self.idrac, license_service_url)
+    license_url = license_service_resp.get('Licenses', {}).get('@odata.id', {})
+    return license_url
+
+
+def get_job_status(self, module, export_license_status):
+    res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+    job_tracking_uri = export_license_status.headers.get("Location")
+    job_id = job_tracking_uri.split("/")[-1]
+    job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+    job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(self.idrac, job_uri)
+    job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+    if job_failed:
+        module.exit_json(
+            msg=job_dict.get('Message'),
+            failed=True,
+            job_details=job_dict)
+    return job_dict
+
+
+def get_share_details(module):
+    share_details = {}
+    share_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    share_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+    share_details["UserName"] = module.params.get('share_parameters').get('username')
+    share_details["Password"] = module.params.get('share_parameters').get('password')
+    return share_details
+
+
+def get_proxy_details(module):
+    proxy_details = {}
+    if module.params.get('share_parameters').get('share_type') == "http":
+        proxy_details["ShareType"] = "HTTP"
+    else:
+        proxy_details["ShareType"] = "HTTPS"
+    proxy_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    proxy_details["ShareName"] = module.params.get('share_parameters').get('share_name')
```

Comment:
Just a suggestion

Have pre-defined dictionary for key:value pair like {'IPAddress': 'ip_address'} and use **kwargs in get_share_details, run the for loop inside or lambda function in get_share_details, it'll be dynamic and easier to assign the values. You can use in almost all the places you need to pass only those values which you want to assign, return the dictionary and use that.

Will reduce lots of repetitive code and more readable.

Reply:
Proxy parameters and share details are kept as separate common function. No code repetitive here. so, no changes done here